src/App.js

```diff
@@ -1,5 +1,105 @@
+import { Console, Random } from "@woowacourse/mission-utils";
+
 class App {
-  async play() {}
+  constructor() {
+    this.answerNumber = '';
+    this.playerInput = '';
+  }
+  // error 핸들링 메서드
+  handleError(message) {
+    let error = new Error();
+    error.message = message;
+    Console.print(error.message);
+    throw error;
+  }
+  // 랜덤 숫자 생성 메서드
+  setAnswerNumber() {
+    const numbers = [];
+    while (numbers.length < 3) {
+      const number = Random.pickNumberInRange(1, 9);
+      if (!numbers.includes(number)) {
+        numbers.push(number);
+      }
+    }
+    this.answerNumber = numbers.join('');
+  }
+  // 입력값이 검증 메서드
+  validateInput(input) {
+    const isDuplicate = input.split('').some((number) => input.indexOf(number) !== input.lastIndexOf(number))
+
+    if(isNaN(Number(input))) {
+      this.handleError('[ERROR] 숫자를 입력해주세요');
+    } else if(input.length !== 3) {
+      this.handleError('[ERROR] 3자리 숫자를 입력해주세요');
+    } else if(input.includes('0')) {
+      this.handleError('[ERROR] 1~9 사이의 숫자를 입력해주세요');
+    }  else if(isDuplicate) {
+      this.handleError('[ERROR] 중복되지 않는 숫자를 입력해주세요');
+    } else {
+      this.playerInput = input;
+    }
+  } 
+  // 볼, 스트라이크 판단 메서드
+  calculateBallsAndStrikes() {
+    const player = this.playerInput.split('');
+    let strike = 0;
+    let ball = 0;
+    player.forEach((number, index) => {
+      if (this.answerNumber.includes(number)) {
+        if (this.answerNumber[index] === player[index]) {
+          strike++;
+        } else {
+          ball++;
+        }
+      }});
+    return { ball, strike };
+  }
+  // 힌트 출력 메서드
+  printHintText(ball, strike) {
+    const ballCount = ball > 0 ? `${ball}볼 ` : '';
+    const strikeCount = strike > 0 ? `${strike}스트라이크` : '';
+    if (strike === 0 && ball === 0) {
+      Console.print('낫싱');
+    } else {
+      Console.print(`${ballCount}${strikeCount}`);
+    }
+  }
+  // 게임 종료 후 재시작 여부 확인
+  async askReplay() {
+    const replay = await Console.readLineAsync('게임을 새로 시작하려면 1, 종료하려면 2를 입력하세요.');
+    if (replay === '1') {
```

Comment:
상수들은 상수화하여 더 쉽게 관리할 수 있습니다.
이름을 주어서 어떤 역할을 하는 지 직관적으로 이해할 수도 있어요!

'1'과 '2'같은 경우 아래처럼 만들 수 있을 것 같네요.
```js
const GAME = {
	REPLAY: '1',
	QUIT: '2',
};
```

```suggestion
    if (replay === GAME.REPLAY) {
```